src/test/scala/vsys/blockchain/state/opcdiffs/AssertOpcDiffTest.scala

```diff
@@ -0,0 +1,94 @@
+
+    AssertOpcDiff.gtEq0(DataEntry(Longs.toByteArray(1), DataType.Amount)) should be (Right(OpcDiff.empty))
+    AssertOpcDiff.gtEq0(DataEntry(Longs.toByteArray(0), DataType.Amount)) should be (Right(OpcDiff.empty))
+    AssertOpcDiff.gtEq0(DataEntry(Longs.toByteArray(-1), DataType.Amount)) should be (
+      Left(GenericError(s"Invalid Assert (gteq0): " +
+        s"Value ${DataEntry(Longs.toByteArray(-1), DataType.Amount).json} is negative")))
+
+    AssertOpcDiff.ltEq(DataEntry(Longs.toByteArray(1), DataType.Amount),
+      DataEntry(Longs.toByteArray(1), DataType.Amount)) should be (Right(OpcDiff.empty))
+    AssertOpcDiff.ltEq(DataEntry(Longs.toByteArray(1), DataType.Amount),
+      DataEntry(Longs.toByteArray(1), DataType.ShortBytes)) should be (Left(ContractDataTypeMismatch))
+    AssertOpcDiff.ltEq(DataEntry(Longs.toByteArray(1), DataType.Amount),
+      DataEntry(Longs.toByteArray(0), DataType.Amount)) should be (
+      Left(GenericError(s"Invalid Assert (lteq0): " +
+        s"Value ${DataEntry(Longs.toByteArray(1), DataType.Amount).json} " +
+        s"is larger than ${DataEntry(Longs.toByteArray(0), DataType.Amount).json}")))
+
+    AssertOpcDiff.ltInt64(DataEntry(Longs.toByteArray(1), DataType.Amount)) should be (Right(OpcDiff.empty))
+    AssertOpcDiff.ltInt64(DataEntry(Longs.toByteArray(0), DataType.ShortBytes)) should be (
+      Left(ContractDataTypeMismatch))
+
+    AssertOpcDiff.gt0(DataEntry(Longs.toByteArray(1), DataType.Amount)) should be (Right(OpcDiff.empty))
+    AssertOpcDiff.gt0(DataEntry(Longs.toByteArray(0), DataType.Amount)) should be (
+      Left(GenericError(s"Invalid Assert (gt0): " +
+        s"Value ${DataEntry(Longs.toByteArray(0), DataType.Amount).json} is non-positive")))
+    AssertOpcDiff.gt0(DataEntry(Longs.toByteArray(-1), DataType.Amount)) should be (
+      Left(GenericError(s"Invalid Assert (gt0): " +
+        s"Value ${DataEntry(Longs.toByteArray(-1), DataType.Amount).json} is non-positive")))
+
+    AssertOpcDiff.equal(DataEntry(Longs.toByteArray(1), DataType.Amount),
+      DataEntry(Longs.toByteArray(1), DataType.Amount)) should be (Right(OpcDiff.empty))
+    AssertOpcDiff.equal(DataEntry(Longs.toByteArray(1), DataType.Amount),
+      DataEntry(Ints.toByteArray(1), DataType.Int32)) should be (
+      Left(GenericError(s"Invalid Assert (eq): DataEntry " +
+        s"${DataEntry(Longs.toByteArray(1), DataType.Amount).json} " +
+        s"is not equal to ${DataEntry(Ints.toByteArray(1), DataType.Int32).json}")))
+
+    val tx = RegisterContractTransaction.create(PrivateKeyAccount(Array.fill(TransactionParser.KeyLength)(0)),
+      ContractPermitted.contract, Seq(DataEntry(Longs.toByteArray(-1), DataType.Amount)),
+      "vsys", 10000L, 100, 1L)
+
+    AssertOpcDiff.isCallerOrigin(ExecutionContext.fromRegConTx(state, TestFunctionalitySettings.Enabled, Option(0L),
+      1L, 1, tx.right.get).right.get)(
```

Comment:
all `ExecutionContext.fromRegConTx(state, TestFunctionalitySettings.Enabled, Option(0L), 1L, 1, tx.right.get).right.get` can be picked out as value context to simplify the code